modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerSim.scala

```diff
@@ -3,24 +3,89 @@
 
 package seqexec.server.gmos
 
+import cats._
+import cats.implicits._
 import cats.effect.Sync
 import cats.effect.Timer
+import cats.effect.concurrent.Ref
+import monocle.Optional
+import monocle.macros.Lenses
+import monocle.std.option.some
 import seqexec.model.dhs.ImageFileId
 import seqexec.model.enum.ObserveCommandResult
 import seqexec.server.InstrumentSystem.ElapsedTime
-import seqexec.server.gmos.GmosController.{GmosConfig, NorthTypes, SiteDependentTypes, SouthTypes}
-import seqexec.server.{InstrumentControllerSim, Progress}
+import seqexec.server.gmos.GmosController.GmosConfig
+import seqexec.server.gmos.GmosController.NorthTypes
+import seqexec.server.gmos.GmosController.SiteDependentTypes
+import seqexec.server.gmos.GmosController.SouthTypes
+import seqexec.server.gmos.GmosController.Config.NSConfig
+import seqexec.server.InstrumentControllerSim
+import seqexec.server.Progress
 import squants.Time
+import scala.concurrent.duration._
 
-object GmosControllerSim {
-  def apply[F[_]: Sync: Timer, T <: SiteDependentTypes](name: String): GmosController[F, T] =
-    new GmosController[F, T] {
-      private val sim: InstrumentControllerSim[F] = InstrumentControllerSim[F](s"GMOS $name")
+@Lenses
+final case class NSCurrent(
+  fileId:        ImageFileId,
+  exposureCount: Int,
+  expTime:       Time
+)
+
+/**
+ * Used to keep the internal state of NS
+ */
+@Lenses
+final case class NSObsState(config: NSConfig, current: Option[NSCurrent])
 
-      override def observe(fileId: ImageFileId, expTime: Time): F[ObserveCommandResult] =
-        sim.observe(fileId, expTime)
+object NSObsState {
+  def fromConfig(c: NSConfig): NSObsState =
+    NSObsState(c, None)
 
-      override def applyConfig(config: GmosConfig[T]): F[Unit] = sim.applyConfig(config)
+  val Zero: NSObsState = NSObsState(NSConfig.NoNodAndShuffle, None)
+
+  val fileId: Optional[NSObsState, ImageFileId] =
+    NSObsState.current ^<-? some ^|-> NSCurrent.fileId
+
+  val exposureCount: Optional[NSObsState, Int] =
+    NSObsState.current ^<-? some ^|-> NSCurrent.exposureCount
+
+  val expTime: Optional[NSObsState, Time] =
+    NSObsState.current ^<-? some ^|-> NSCurrent.expTime
+}
+
+object GmosControllerSim {
```

Comment:
The simulator got a lot more complex to support N&S. It needs to keep an internal state